From cb854af0a92e82e4640df0a8152280b9c9628da8 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Thu, 11 Oct 2007 22:42:04 +0800
Subject: [PATCH] [EXPAND] Add likely flags in expari

The case where the expansion isn't quoted is the norm.
---
 ChangeLog    |    1 +
 src/expand.c |    4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 1f26b83..98c546e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -2,6 +2,7 @@
 
 	* Add assignment support in arithmetic expansions.
 	* Size optimisations in arithmetic lexer.
+	* Add likely flags in expari.
 
 2007-10-08  Herbert Xu <herbert@gondor.apana.org.au>
 
diff --git a/src/expand.c b/src/expand.c
index 9cb8eab..5c31400 100644
--- a/src/expand.c
+++ b/src/expand.c
@@ -523,7 +523,7 @@ expari(int flag)
 
 	expdest = p;
 
-	if (flag & QUOTES_ESC)
+	if (likely(flag & QUOTES_ESC))
 		rmescapes(p + 1);
 
 	result = arith(p + 1);
@@ -531,7 +531,7 @@ expari(int flag)
 
 	len = cvtnum(result);
 
-	if (!(flag & EXP_QUOTED))
+	if (likely(!(flag & EXP_QUOTED)))
 		recordregion(begoff, begoff + len, 0);
 }
 
-- 
1.5.5

